test/runtests.jl

```diff
@@ -10,81 +11,201 @@ using Test
+                    @info "Failing root" root=r expected=expected_root initial_guess=x0 tol=_default_tol
+                end
+            end
+            @test all(abs.(roots .- expected_root) .< _default_tol)
+        else
+            if abs(roots - expected_root) ≥ _default_tol
+                @info "Failing root" root=roots expected=expected_root initial_guess=problem.x_init tol=_default_tol
+            end
+            @test abs(roots - expected_root) < _default_tol
+        end
+    elseif tol isa ResidualTolerance
+        # For residual tolerance, check that function values are small
+        if roots isa AbstractArray
+            @test all(map(x -> abs(problem.f(x)), roots) .< tol.tol)
+        else
+            @test abs(problem.f(roots)) < tol.tol
+        end
+    elseif tol isa SolutionTolerance
+        # For solution tolerance, check that roots are close to expected
+        if roots isa AbstractArray
+            @test all(abs.(roots .- expected_root) .< tol_factor * tol.tol)
+        else
+            @test abs(roots - expected_root) < tol_factor * tol.tol
+        end
+    elseif tol isa RelativeSolutionTolerance
+        # For relative tolerance, check relative difference
+        # Avoid division by zero
+        if abs(expected_root) < eps(typeof(expected_root))
+            if roots isa AbstractArray
+                @test all(abs.(roots .- expected_root)) .< 50*tol.tol
+            else
+                @test abs(roots - expected_root) < tol_factor * tol.tol
+            end
+        else
+            if roots isa AbstractArray
+                @test all(abs.((roots .- expected_root) ./ expected_root) .< tol_factor * tol.tol)
+            else
+                @test abs((roots - expected_root) / expected_root) < tol_factor * tol.tol
+            end
+        end
+    elseif tol isa RelativeOrAbsoluteSolutionTolerance
+        # For combined tolerance, check both relative and absolute
+        if roots isa AbstractArray
+            @test all((abs.((roots .- expected_root) ./ expected_root) .< tol_factor * tol.rtol) .| (abs.(roots .- expected_root) .< tol_factor * tol.atol))
+        else
+            @test (abs((roots - expected_root) / expected_root) < tol_factor * tol.rtol) || (abs(roots - expected_root) < tol_factor * tol.atol)
+        end
+    else
+        # Default tolerance check - use a reasonable default based on floating point precision
+        FT = typeof(expected_root)
+        _default_tol = tol_factor * default_tol(FT).tol  
+        if roots isa AbstractArray
+            @test all(abs.(roots .- expected_root) .< _default_tol)
+        else
+            @test abs(roots - expected_root) < _default_tol
+        end
+    end
+end
+
+# Helper function to get maxiters based on problem name
+function get_maxiters(problem_name)
+    if problem_name in ["high-multiplicity root", "steep exponential function", "trigonometric function"]
+        return 10_000
+    else
+        return 10_000
+    end
+end
```

Comment:
is it intentional to have the same iters in every case?

Reply:
Leftover from experimentation. I removed the if ... else.